### DISCUSSION TITLE
Binding problems in custom control with ContentView

### DISCUSSION THREAD

**Discussion by TonyK-FmGroup** [dotnet/maui#7623] [Q&A] [answered]
Hi,

I have two near-identical user controls using `ContentView` which contain a `CollectionView` provided with a `BindableProperty`  each of type `ObservableCollection<T>` where 'T' is a different type in each control. 

In one of them I need to use `BindingContext = this` for it to correctly see the properties in the code behind, in the other adding this breaks the bindings. Having checked the code in each very carefully, even going as far as to copy/paste to ensure it's the same, I still see this behaviour. 

With the one that needs the BindingContext set, this one also does not see updates from the view model when the underlying property changes. Again, the property is created in exactly the same way as the other, the property changed event raise in exactly the same way. 

I feel I've missed something, anything obvious?   I'll try to create an example in a separate project and post some code soon.

EDIT
The issue with the BindingContext appears to be where the property names in the custom control are the same as in the main page context (my view model) then those pass through including the change events, but, where the property names in the custom control are different to those in the view model, I need to set the context, but, the change events do not update the values in the control.

Sample code here:  [https://github.com/TonyK-FmGroup/MauiBindingDemo](https://github.com/TonyK-FmGroup/MauiBindingDemo)
See that one of the custom controls happily shows the data, the other does not. I also found that I do not see binding failures on the custom control even if i use totally bogus source names for the ItemsSource.

**Comment by iot-crazy** [comment, marked as answer]
I found this article that solved the problem. As I suspected, I just misunderstood how the binding works in MAUI 😀

[https://xamgirl.com/tips-and-ticks-when-creating-a-custom-control-in-xamarin-forms-part-1/](https://xamgirl.com/tips-and-ticks-when-creating-a-custom-control-in-xamarin-forms-part-1/)
